pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,192 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useContext, useEffect, useState } from 'react';
+import cockpit from 'cockpit';
+import { Button } from '@patternfly/react-core/dist/esm/components/Button/index.js';
+import { ClipboardCopy } from '@patternfly/react-core/dist/esm/components/ClipboardCopy/index.js';
+import { EmptyState, EmptyStateBody } from '@patternfly/react-core/dist/esm/components/EmptyState/index.js';
+import { FormGroup, FormFieldGroup, FormFieldGroupHeader } from '@patternfly/react-core/dist/esm/components/Form/index.js';
+import { Grid } from '@patternfly/react-core/dist/esm/layouts/Grid/index.js';
+import { InputGroup } from '@patternfly/react-core/dist/esm/components/InputGroup/index.js';
+import { MinusIcon } from '@patternfly/react-icons';
+import { TextInput } from '@patternfly/react-core/dist/esm/components/TextInput/index.js';
+
+import { Name, NetworkModal, dialogSave } from "./dialogs-common";
+import { ModelContext } from './model-context';
+import { useDialogs } from 'dialogs.jsx';
+
+import './wireguard.scss';
+
+const _ = cockpit.gettext;
+
+export function WireGuardDialog({ settings }) {
+    const Dialogs = useDialogs();
+    const idPrefix = 'network-wireguard-settings';
+    const model = useContext(ModelContext);
+
+    const [iface, setIface] = useState(settings.connection.interface_name);
+    const [privateKey, setPrivateKey] = useState('');
+    const [publicKey, setPublicKey] = useState('');
+    const [listenPort, setListenPort] = useState('');
+    const [addresses, setAddresses] = useState('');
+    const [dialogError, setDialogError] = useState('');
+    const [peers, setPeers] = useState([]);
+
+    useEffect(() => {
+        async function getPublicKey() {
+            try {
+                const key = await cockpit.script(`echo ${privateKey.trim()} | wg pubkey`, { err: 'message' });
```

Comment:
This will appear in `ps` and /proc. Please never put secrets into the command line. Instead, write it into stdin directly, with something like `cockpit.spawn(["wg", "pubkey"]).input(privateKey.trim())`. See https://cockpit-project.org/guide/latest/cockpit-spawn.html